tests/test_linalg.py

```diff
@@ -133,8 +133,8 @@ def test_dgemm_vs_scipy(alpha,
 
 def test_dgemm_input_handling():
     alpha = 1.0
-    view_a = pk.from_numpy(np.zeros((4, 3)))
-    view_b = pk.from_numpy(np.zeros((4, 3)))
+    view_a = pk.from_array(np.zeros((4, 3)))
+    view_b = pk.from_array(np.zeros((4, 3)))
```

Comment:
these changes are good, but we should test with something else that implements the protocol

Reply:
I can add a 1D array test? the 2D arrays convert badly with lists because the types get weird when you have lists of lists.

Reply:
Let's start with 1D

Reply:
got around the list of lists problems so did not write a 1D example here